imap_processing/cli.py

```diff
@@ -831,6 +832,46 @@ def do_processing(
         return datasets
 
 
+class Spacecraft(ProcessInstrument):
+    """Process Spacecraft data."""
+
+    def do_processing(
+        self, dependencies: ProcessingInputCollection
+    ) -> list[xr.Dataset]:
+        """
+        Perform Spacecraft specific processing.
+
+        Parameters
+        ----------
+        dependencies : ProcessingInputCollection
+            Object containing dependencies to process.
+
+        Returns
+        -------
+        datasets : xr.Dataset
+            Xr.Dataset of products.
+        """
+        print(f"Processing Spacecraft {self.data_level}")
+
+        if self.data_level != "l1a":
+            raise NotImplementedError(
+                f"Spacecraft processing not implemented for level {self.data_level}"
+            )
+
+        # File path is expected output file path
+        input_files = dependencies.get_file_paths(source="spacecraft")
```

Comment:
I think it's this that you want. Then move this line below before passing it to the code.
```suggestion
        input_files = dependencies.processing_input
```

Reply:
I'm not sure I follow what you mean. I want the file path directly and don't really care about processing input I don't think? i.e. I don't care what type of input it is for this processing, so I am just skipping the processing_input separation into science/ancillary/spice files because it is unnecessary for this instrument. I might be missing what processing_input is used for.

Reply:
Going to merge this so I can get it in before MSIM. Feel free to open an issue for follow-up on this and I'm happy to address it then.